File: guess.py
# import required dependencies
import random

# generate a random number between 1 and 10
# this is the number the user must guess
num = random.randint(1,10)

# declare an empty variable called "guess"
guess = None

# while the user's guess is not the random number from above, loop
while guess != num:

	# get a guess from the user
    guess = input("guess a number between 1 and 10: ")
    # make sure that guess is an integer
    guess = int(guess)

    # the two lines above could be a single line:
    # guess = int(input("guess a number between 1 and 10: "))

    # if the user guessed correctly
    if guess == num:
        print(f"congratulations! you won! {num} was the winner!")
	# if the user guessed too high
    elif guess > num:
        print(f"nope, sorry, {guess} is too high!")
    # if the user guessed too low
    elif guess < num:
        print(f"nope, sorry, {guess} is too low!")

File: Guess.java
// if you run this from an IDE, you'll use a package like the line below
// package webinar.examples;
// if you're running this from the CLI, leave the line above commented


// import required dependencies
import java.util.Random;
import java.util.Scanner;

/*
    Create Java class Guess. This class will play a small and simple
    "Guess My Number Game". It is meant as a very simple introduction
    to working with Java. 
*/
public class Guess {

    // Declare the main() method - (this is where the application starts)
    public static void main(String[] args) {

        // Initialize an object from the Random class
        Random random = new Random();
        // Initialize a randon in - this will be the number the player 
        // is trying to guess
        int num = random.nextInt(10);

        // Initialize a simple boolean to track whether or not 
        // to continue playing
        boolean continuePlaying = true;

        // Communicate with the user 
        System.out.println("Welcome to GUESS MY NUMBER!!");
        System.out.println("Ok, I've got a number in mind...");
        
        // While we should continue playing, loop
        while(continuePlaying) {

            // Prompt the user for their guess
            System.out.println("Please enter a number between 1 & 10: ");
            // Initialize an object of the Scanner class
            Scanner scanner = new Scanner(System.in);
            // Use the scanner to read in the user's guess
            int guess = scanner.nextInt();

            // If the user guessed correctly
            if (guess == num){
                System.out.println("You win! " + guess + " was the right number!");
                
                // STOP THE GAME - looping will cease
                continuePlaying = false;
            } 
            // If the user's guess is too high
            else if (guess > num){
                System.out.println("Nope, sorry, " + guess + " is too high!");
            } 
            // If the user's guess is too low
            else if (guess < num){
                System.out.println("Nope, sorry, " + guess + " is too low");
            }
        }
    }
}

